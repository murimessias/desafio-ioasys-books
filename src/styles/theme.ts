export default {
  border: {
    radius: {
      md: '0.4rem',
      rounded: '9999px',
    },
  },

  colors: {
    white: '#FFFFFF',
    caramel: '#F5F5EE',
    black: '#333333',
    gray: '#999999',
    accent: '#AB2680',
  },

  font: {
    family:
      "Heebo, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",

    sizes: {
      sm: '1.2rem',
      md: '1.4rem',
      lg: '1.6rem',
      xl: '2.8rem',
    },

    weight: {
      normal: 400,
      bold: 500,
    },
  },

  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },

  leading: {
    none: 1,
    tight: 1.25,
    normal: 1.5,
  },

  spacings: {
    xs: '1.6rem',
    sm: '2.4rem',
    md: '3.2rem',
    lg: '4.0rem',
    xl: '4.8rem',
  },
} as const
