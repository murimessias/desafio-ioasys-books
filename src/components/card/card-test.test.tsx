import { screen } from '@testing-library/react'
import { renderWithTheme } from 'resources/utils/tests/helpers'
import theme from 'styles/theme'

import { Card } from '.'

const cardProps = {
  onClick: jest.fn(),
  title: 'Hooked',
  authors: ['Nir Eyal', 'Ryan Hoover'],
  imageUrl: '/img/book-img-placeholder.png',
  pageCount: 150,
  publisher: 'Loyola',
  published: 2020,
}

describe('<Card />', () => {
  it('should render the card correctly', () => {
    const { container } = renderWithTheme(<Card {...cardProps} />)

    expect(screen.getByRole('heading', { name: /hooked/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the data correctly', () => {
    renderWithTheme(<Card {...cardProps} />)

    expect(screen.getByRole('img')).toBeInTheDocument()

    const mediumFontSize = theme.font.sizes.md
    expect(screen.getByRole('heading', { name: /hooked/i })).toHaveStyle({
      'font-size': mediumFontSize,
    })

    const colorAccent = theme.colors.accent
    expect(screen.getByText('Nir Eyal')).toHaveStyle({
      color: colorAccent,
    })
  })
})
