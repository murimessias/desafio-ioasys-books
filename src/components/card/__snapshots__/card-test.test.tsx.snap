// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the card correctly 1`] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFFFFF;
  border-radius: 0.4rem;
  box-shadow: 0 0.6rem 2.4rem rgba(84,16,95,0.15);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 16rem;
  padding: 1.6rem 0.8rem;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  width: 100%;
}

.c0:hover {
  box-shadow: 0 0.6rem 2.4rem rgba(84,16,95,0.35);
}

.c0:active {
  box-shadow: 0 0.3rem 1.2rem rgba(84,16,95,0.15);
}

.c1 {
  display: block;
  margin-right: 1.6rem;
  max-width: 10rem;
  position: relative;
  width: 100%;
}

.c1 img {
  box-shadow: 0 0.4rem 1.2rem rgba(84,16,95,0.25);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  color: #333333;
  font-size: 1.4rem;
}

.c4 {
  color: #AB2680;
  font-weight: 400;
  list-style: none;
  line-height: 1.25;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 0.4rem;
}

.c5 span {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: 0.2rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Hooked"
      layout="fill"
      src="/img/book-img-placeholder.png"
    />
  </div>
  <div
    class="c2"
  >
    <div
      class=""
    >
      <h2
        class="c3"
      >
        Hooked
      </h2>
      <ul
        class=""
      >
        <li
          class="c4"
        >
          Nir Eyal
        </li>
        <li
          class="c4"
        >
          Ryan Hoover
        </li>
      </ul>
    </div>
    <div
      class="c5"
    >
      <span>
        150
         p√°ginas
      </span>
      <span>
        Editora 
        Loyola
      </span>
      <span>
        Publicado em 
        2020
      </span>
    </div>
  </div>
</div>
`
